thefuck/rules/terraform_no_command.py

```diff
@@ -0,0 +1,16 @@
+import re
+from thefuck.utils import for_app
+
+MISTAKE = r"(?<=Terraform has no command named \")(.*)(?=\"\.)"
+FIX = r"(?<=Did you mean \")(.*)(?=\"\?)"
+
+
+@for_app('terraform')
+def match(command):
+    return ('Terraform has no command named' in command.output and 'Did you mean' in command.output)
+
+
+def get_new_command(command):
+    mistake = re.search(MISTAKE, command.output).group(0)
+    fix = re.search(FIX, command.output).group(0)
```

Comment:
These lines can fail if the structure of the output message changes a bit — such as punctuation. If that ever happens, the rule will start failing. But that's better than it silently stop matching because of a stricter `match` function.

That hints at the possibility of running each rule inside a try/except block and displaying a proper error instead of letting it explode. But I digress (note taken).

Reply:
Ok, I changed the `match` method to only return positive if both regex patterns can be matched against the command output, so `get_new_command` hypothetically won't ever fail.

Reply:
I'm not sure whether I'd prefer having it fail or not. This way it wouldn't silently fail to match if the output ever changes in the future.

The failure would have to be properly handled (try/except) — which can be done separately. With that, I think it should fail. The failure will be logged and thefuck would jump to the next rule.

Without it being handled, which is the current behavior, thefuck would crash and any following rules would not be attempted ( namely their `get_new_command`).

To me, a failing rule is way more noticeable that a silent, non-matching one.

WDYT?

Reply:
I think tracking a moving target can be tough - just how lenient should the `match` function be in order for the rule to continue matching when the output changes? It could be problematic to ask the writers of the rules to come up with a match function that is broad enough to capture possible evolutions of the error message while remaining specific to the problem it is addressing. 

My thought process in the moment was that 'if the rule matches then `fuck` should be able to provide the replacement command'. Either way, I would be happy to revert back to the original matching logic if you feel strongly about it